e2e/pkg/wait.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+//+build e2e
+
+package pkg
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/pkg/errors"
+)
+
+// WaitConfig contains configuration for WaitForFunc.
+type WaitConfig struct {
+	Timeout        time.Duration
+	Interval       time.Duration
+	TolerateErrors int
+	Logger         logrus.FieldLogger
+}
+
+// NewWaitConfig create new WaitConfig.
+func NewWaitConfig(timeout, interval time.Duration, tolerateErrs int, log logrus.FieldLogger) WaitConfig {
+	return WaitConfig{
+		Timeout:        timeout,
+		Interval:       interval,
+		TolerateErrors: tolerateErrs,
+		Logger:         log,
+	}
+}
```

Comment:
One other option for this is to setup a webhook receiving channel like I did with `ctest`. Probably good to consider later in the broader E2E framework planning.